Sources/LanguageServerProtocol/SupportTypes/TextDocumentEdit.swift

```diff
@@ -52,4 +52,31 @@ public struct TextDocumentEdit: Hashable, Codable, Sendable {
     self.textDocument = textDocument
     self.edits = edits
   }
+
+  public enum CodingKeys: String, CodingKey {
+    case kind
+    case textDocument
+    case edits
+  }
+
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode("textDocumentEdit", forKey: .kind)
+    try container.encode(self.textDocument, forKey: .textDocument)
+    try container.encode(self.edits, forKey: .edits)
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    let kind = try container.decode(String.self, forKey: .kind)
```

Comment:
Where is the `kind` and `textDocumentEdit` coming from? I can’t see it mentioned in the LSP spec anywhere.

Reply:
Ah, good catch. I added these based on `CreateFile`, but the actual issue was somewhere else ([OptionalVersionedTextDocumentIdentifier](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#optionalVersionedTextDocumentIdentifier)). I've removed these.

Reply:
Done in https://github.com/apple/sourcekit-lsp/pull/1193/commits/4c1e119197f5997af6d9cc6e6f5a404d09e4e470